File: backend/src/main/java/group8/bloodbank/model/Anamnesis.java
package group8.bloodbank.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.checkerframework.checker.units.qual.C;

import javax.persistence.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Anamnesis {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private double weight;
    @Column
    private boolean sicknessSymptoms;
    @Column
    private boolean skinIllness;
    @Column
    private boolean bloodPressureAbnormalities;
    @Column
    private boolean therapyIntake;
    @Column
    private boolean menstrualCycle;
    @Column
    private boolean dentalInterventions;
    @Column
    private boolean skinPiercings;
}

File: frontend/src/app/model/anamnesis.ts
export class Anamnesis {
    id!: number;
    weight!: number;
    sicknessSymptoms!: boolean;
    skinIllness!: boolean;
    bloodPressureAbnormalities!: boolean;
    therapyIntake!: boolean;
    menstrualCycle!: boolean;
    dentalInterventions!: boolean;
    skinPiercings!: boolean;

  public constructor(obj?: any) {
      if (obj) {
          this.id = obj.id;
          this.bloodPressureAbnormalities = obj.bloodPressureAbnormalities;
          this.therapyIntake = obj.therapyIntake;
          this.weight = obj.weight;
          this.skinIllness = obj.skinIllness;
          this.menstrualCycle = obj.menstrualCycle;
          this.dentalInterventions = obj.dentalInterventions;
          this.skinPiercings = obj.skinPiercings;
          this.sicknessSymptoms = obj.sicknessSymptoms;
      }
  }
}
